From de69815466c4b9ccfe2faaec6a84f4d133f086d7 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 20 Dec 2011 12:15:01 +0800
Subject: [PATCH 187/249] ENGR00170744 IPU lib:Control interface supports
 allocating pmem buf

This patch supports to allocate and free pmem buf via task control
interface for Android.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index afa32cc..b1d828b 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -3509,16 +3509,29 @@ int mxc_ipu_lib_task_control(int ctl_cmd, void * arg, ipu_lib_handle_t * ipu_han
 	{
 		ipu_lib_ctl_mem_t * ctl_mem_info =
 				(ipu_lib_ctl_mem_t *) arg;
+#ifdef BUILD_FOR_ANDROID
+		ret = __pmem_alloc(&ctl_mem_info->minfo,
+				   &ctl_mem_info->mmap_vaddr);
+		if (ret == 0)
+			memset(ctl_mem_info->mmap_vaddr, 0,
+			       ctl_mem_info->minfo.size);
+#else
 		ret = __ipu_mem_alloc(&ctl_mem_info->minfo,
 				&ctl_mem_info->mmap_vaddr);
+#endif
 		break;
 	}
 	case IPU_CTL_FREE_MEM:
 	{
 		ipu_lib_ctl_mem_t * ctl_mem_info =
 				(ipu_lib_ctl_mem_t *) arg;
+#ifdef BUILD_FOR_ANDROID
+		 __pmem_free(&ctl_mem_info->minfo,
+			     &ctl_mem_info->mmap_vaddr);
+#else
 		 __ipu_mem_free(&ctl_mem_info->minfo,
 				&ctl_mem_info->mmap_vaddr);
+#endif
 		break;
 	}
 	case IPU_CTL_TASK_QUERY:
-- 
1.8.0

